src/components/util/remote-ressource-handler.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+export class RemoteRessourceHandler {
+    fetcher = undefined;
+    errorHandler = undefined;
+    postUpdate = undefined;
+
+    constructor(fetcher, errorHandler, postUpdate) {
+        this.fetcher = fetcher;
+        this.errorHandler = errorHandler;
+        this.postUpdate = postUpdate;
+    }
+
+    values = undefined;
+    updating = undefined;
+
+    cbUpdateDone = new Set();
+
+    get(cbUpdateDone) {
```

Comment:
I think this API is a bit weird.. always return a promise, instead of returning undefined or the value, and calling the callback ? Or if we stick with the current API, rename get to getOrFetch ? getOtherwiseFetch ?